### PR TITLE
Tinygo codegen v2

### PR DESCRIPTION
**Pull request by stevelr**
This PR fixes a tinygo codegen bug for structs with optional fields, and adds cbor generation.

### CODE REVIEW

File: build/update-tinygo.sh [hunk 1]
```diff
@@ -9,11 +9,13 @@
...
-actor_sdk_version=3893510790129936fa561b386eedcec6c71338a0
-tinygo_msgpack_version=192ec93ec5c440e685d61c705c8fe68ff4aa1973
+actor_sdk_version=v0.1.1
+tinygo_msgpack_version=v0.1.4
+tinygo_cbor_version=v0.1.0
 
 WASH=${WASH:-wash}
-$WASH gen -c ./codegen-go.toml
+#$WASH gen -c ./codegen-go.toml
+../weld/target/debug/codegen ./codegen-go.toml
```
Comment (brooksmtownsend) [inline build/update-tinygo.sh:18]: Is this intended to still be just the debug weld or should it be back to `$WASH`?

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
just the tiniest nit

File: build/update-tinygo.sh (hunk 1 above)
Comment (stevelr) [reply build/update-tinygo.sh:18]: It won't work with wash yet until wash updates it's codegen dependency to 0.4.5 and has a new release

**Review by brooksmtownsend — APPROVED** [review]

File: build/update-tinygo.sh (hunk 1 above)
Comment (brooksmtownsend) [reply build/update-tinygo.sh:18]: Okay, I'm fine with this since it's a script we're likely only using internally

**Review by autodidaddict — APPROVED** [review]

**Pull request merged**